.github/workflows/publish.yaml

```diff
@@ -126,4 +127,11 @@ jobs:
             cmd="closeAndReleaseSonatypeStagingRepository";
           fi
           echo "Publishing Version $VERSION to Sonatype"
-          ./gradlew publishToSonatype ${cmd} --no-parallel -Pversion=$VERSION -Psigning.gnupg.executable=gpg -Psigning.gnupg.passphrase="${{ secrets.ORG_GPG_PASSPHRASE }}"
\ No newline at end of file
+          ./gradlew publishToSonatype ${cmd} --no-parallel -Pversion=$VERSION -Psigning.gnupg.executable=gpg -Psigning.gnupg.passphrase="${{ secrets.ORG_GPG_PASSPHRASE }}"
+
+  publish-to-swaggerhub:
+    name: "Publish OpenAPI spec to Swaggerhub"
+    permissions:
+      contents: read
+    needs: [ secret-presence ]
```

Comment:
please add the 
```yaml
if: needs.secret-presence.output.HAS_SWAGGER
```
otherwise this job will always run. `needs` only expresses dependency and ordering.